packages/flutter_test/lib/src/binding.dart

```diff
@@ -485,6 +496,7 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
     final Zone testZone = _parentZone.fork(specification: errorHandlingZoneSpecification);
     testZone.runBinary(_runTestBody, testBody, invariantTester)
       .whenComplete(testCompletionHandler);
+    timeout?.whenComplete(testCompletionHandler);
```

Comment:
This won't cause the test to fail, right?  Can we add a test to verify that home-growth timeout logic will correctly cause a slow test to fail?